File: Rust/beginning/leetcode/11.rs
struct Solution {}
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut res = 0;
        let (mut start, mut end) = (0, height.len() - 1);
        while start < end {
            let d = end - start;
            let h: i32;
            if height[start] < height[end] {
                h = height[start];
                start += 1;
            } else {
                h = height[end];
                end -= 1;
            }
            let tmp = d as i32 * h;
            if res < tmp {
                res = tmp;
            }
        }
        return res;
    }
}

File: code/Python/leetcode/11.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        res: int = 0
        length = len(height)
        (start, end) = (0, length - 1)
        while start < end:
            d = end - start
            h = 0
            if height[start] < height[end]:
                h = height[start]
                start += 1
            else:
                h = height[end]
                end -= 1
            tmp = d * h
            if res < tmp:
                res = tmp
        return res


if __name__ == "__main__":
    num1 = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    print("结果为：", Solution.maxArea(Solution, height=num1), end="")
